src/main/java/frc/robot/systems/GrabberFSM.java

```diff
@@ -0,0 +1,233 @@
+package frc.robot.systems;
+
+import com.revrobotics.CANSparkMax;
+
+import edu.wpi.first.wpilibj.Joystick;
+// Third party Hardware Imports
+// Robot Imports
+import frc.robot.TeleopInput;
+import frc.robot.HardwareMap;
+
+public class GrabberFSM {
+	/* ======================== Constants ======================== */
+	// FSM state definitions
+	public enum FSMState {
+		LOW_IDLE,
+		ASCENDING,
+        CONTROL,
+        DESCENDING,
+        MID_IDLE
```

Comment:
format - some weird spacing issues throughout file, please fix. check tabs vs spaces if it looks right locally but not here?